src/threadlist.cpp

```diff
@@ -782,6 +823,27 @@ restarthread(void *threadv)
     TLSInfo_SetThreadSysinfo(saved_sysinfo);
   }
 
+  if (thread == motherofall) { // if this is a user thread
+    /* If DMTCP_RESTART_PAUSE==3, sleep 15 seconds to allow gdb attach.*/
+    char * pause_param = getenv("DMTCP_RESTART_PAUSE");
+    if (pause_param == NULL) {
+      pause_param = getenv("MTCP_RESTART_PAUSE");
+    }
+    if (pause_param != NULL && pause_param[0] == '3' &&
```

Comment:
Same comment as above.  If we use `strncmp', then we should use the simpler `strcmp` since there is no safety issue.  But I prefer the C primitives.